towerlib/towerlib.py

```diff
@@ -1650,6 +1650,30 @@ def workflow_job_templates(self):
                              entity_object='JobTemplate',
                              primary_match_field='name')
 
+    def get_workflow_job_template_by_id(self, id_):
+        """Retrieves a workflow template job by id.
+
+        Args:
+            id_: The id of the workflow template job to retrieve.
+
+        Returns:
+            Host: The job if a match is found else None.
+
+        """
+        return next(self.workflow_job_templates.filter({'id': id_}), None)
+
+    def get_workflow_job_templates_by_name(self, name):
+        """Retrieves all workflow template jobs matching a certain name.
+
+        Args:
+            name: The name of the workflow template job(s) to retrieve.
+
+        Returns:
+            UnifiedJob (Generator): A workflow template job generator.
+
+        """
+        return next(self.workflow_job_templates.filter({'name__iexact': name}))
```

Comment:
This will raise an exception if no match is found, defaulting to None would make it safer.

Reply:
I see, but there are also methods that don't return `None`, like    `def get_workflow_jobs_by_name(self, name)`
https://github.com/schubergphilis/towerlib/pull/60/files#diff-983597244fad29c7692a852e06dc5066R1638

Reply:
Very valid point, that also needs fixing if you don't mind  :)